test-suite/test-ast.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+#
+## Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# Should be run from the source root directory.
+# The 'xunused' utility should be placed into the source root directory. 
+
+log=/tmp/test-ast.log
+
+configureBinary=./configure 
+xunusedBinary=./xunused
+xunusedResult=./xunused_result.txt
+
+if [ ! -x $configureBinary ]
+then
+    echo `basename "$0"`": Must be run in the root directory"
+    exit 1
+fi
+
+if [ ! -x $xunusedBinary ]
+then
+    echo "Could not find xunused in the root directory"
+    exit 1
+fi
+
+customCompileCommands=$1
+defaultCompileCommands=compile_commands.json
+compileCommands=./${defaultCompileCommands}
+
+if test -n "$customCompileCommands"
+then
+    compileCommands=$customCompileCommands
```

Comment:
Please refactor to reuse the existing commands file when such reuse is explicitly requested by specifying that file name (rather than to always create that file, with default or custom name).